api/v1alpha1/hardwareclassification_types.go

```diff
@@ -123,18 +123,72 @@ type Ram struct {
 	MaximumSizeGB int `json:"maximumSizeGB,omitempty"`
 }
 
+// ProfileMatchStatus represents the state of the HardwareClassification
+type ProfileMatchStatus string
+
+const (
+	// ProfileMatchStatusEmpty is the default status value
+	ProfileMatchStatusEmpty ProfileMatchStatus = ""
+
+	// ProfileMatchStatusMatched is the status value when the profile
+	// matches to one of the BareMtalHost.
+	ProfileMatchStatusMatched ProfileMatchStatus = "matched"
+
+	// ProfileMatchStatusUnMatched is the status value when the profile
+	// not matches to one of the BareMtalHost.
+	ProfileMatchStatusUnMatched ProfileMatchStatus = "unmatched"
+)
+
+// ErrorType indicates the class of problem that has caused the HCC resource
+// to enter an error state.
+type ErrorType string
+
+const (
+	// LabelUpdateFailure is an error condition occurring when the
+	// controller is unalble to update label of BareMetalHost.
+	LabelUpdateFailure ErrorType = "label update error"
+
+	// LabelDeleteFailure is an error condition occurring when the
+	// controller is unalble to delete label of BareMetalHost.
+	LabelDeleteFailure ErrorType = "label delete error"
+
+	// FetchBMHListFailure is an error condition occurring when the
+	// controller is unable to fetch BMH from BMO
+	FetchBMHListFailure ErrorType = "fetch BMH from BMO error"
```

Comment:
maybe no need for the "from BMO" in here

Reply:
@maelk DONE